adoc/admin-security-certificates.adoc

```diff
@@ -0,0 +1,208 @@
+|Server,Client
+
+|kube-etcd-server
+|etcd-ca
+|etcd/server.crt,key
+|Server,Client
+|===
+
+The following certificates are created by `skuba` and are stored in `Secret` resource.
+
+[%header,cols=4*]
+|===
+|Common Name
+|Parent CA
+|Secret Resource Name
+|Kind
+
+|oidc-dex
+|kubernetes
+|oidc-dex-cert
+|Server
+
+|oidc-gangway
+|kubernetes
+|oidc-gangway-cert
+|Server
+
+|cilium-etcd-client
+|etcd-ca
+|cilium-secret
+|Client
+|===
+
+== Deployment with custom root CA certificate
+
+Administrators can provide custom root CA certificates during cluster deployment and administrators can decide which root CA components to replace or replace all.
+
+After `skuba cluster init`, go to the generated folder. Create a `pki` folder and put your custom root CA certificate into `pki` folder.
+
+* Replace {kube} apiserver root CA certificate
++
+[source,bash]
+----
+mkdir -p my-cluster/pki
+cp <custom-apiserver-rootca-cert-path> my-cluster/pki/ca.crt
+cp <custom-apiserver-rootca-key-path> my-cluster/pki/ca.key
+chmod 644 my-cluster/pki/ca.crt
+chmod 600 my-cluster/pki/ca.key
+----
+
+* Replace front-end proxy root CA certificate
++
+[source,bash]
+----
+mkdir -p my-cluster/pki
+cp <custom-frontproxy-rootca-cert-path> my-cluster/pki/front-proxy-ca.crt
+cp <custom-frontproxy-rootca-key-path> my-cluster/pki/front-proxy-ca.key
+chmod 644 my-cluster/pki/front-proxy-ca.crt
+chmod 600 my-cluster/pki/front-proxy-ca.key
+----
+
+* Replace etcd root CA certificate
++
+[source,bash]
+----
+mkdir -p my-cluster/pki/etcd
+cp <custom-etcd-rootca-cert-path> my-cluster/pki/etcd/ca.crt
+cp <custom-etcd-rootca-key-path> my-cluster/pki/etcd/ca.key
+chmod 644 my-cluster/pki/etcd/ca.crt
+chmod 600 my-cluster/pki/etcd/ca.key
+----
+
+Then bootstrap the cluster `skuba node bootstrap`.
+
+== Automatic certificate renewal
+
+{productname} renews all the certificates automatically during control plane update <<handling_updates>>.
+
+[NOTE]
+====
+It is a best practice to update your {kube} cluster frequently in order to stay secure.
+====
+
+== Manual certificate renewal
+
+. Renewal certificate managed by `kubeadm`
++
+[IMPORTANT]
+====
+If you are running multiple master nodes, administrators need to run the following commands on all the master nodes sequentially.
+====
++
+[source,bash]
+----
+ssh <username>@<master-node-ip-address/fqdn>
+sudo kubeadm alpha certs renew all
+sudo reboot
+----
++
+Copy the renewed `admin.conf` from one of the master nodes to your local environment
++
+[source,bash]
+----
+ssh <username>@<master-node-ip-address/fqdn>
+sudo cat /etc/kubernetes/admin.conf
+----
+
+. Renewal certificate created by `skuba`
++
+Login into master node and regenerate the certificates
+
+* Replace oidc-dex secret
++
+[source,bash]
+----
+cd /etc/kubernetes/pki
+----
++
+Sign the oidc-dex server certificate with root CA `ca.crt/ca.key` and SAN as <control-plane-ip-address/fqdn>. Then update the secret data `ca.crt`, `tls.crt`, and `tls.key` with base64 encoded
```

Comment:
Suggest:

Sign the oidc-dex server certificate with root CA `ca.crt/ca.key` and SAN as <control-plane-ip-address/fqdn>. Then update the {kube} cluster secret data `ca.crt`, `tls.crt`, and `tls.key` with base64 encoded.